{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required namespace: SummarizedExperiment\n",
      "\n",
      "Loading required namespace: HDF5Array\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: 'MatrixGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:parallel':\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:grDevices':\n",
      "\n",
      "    windows\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'Biobase'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:MatrixGenerics':\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "\n",
      "Warning message:\n",
      "\"package 'DelayedArray' was built under R version 4.0.4\"\n",
      "Loading required package: Matrix\n",
      "\n",
      "\n",
      "Attaching package: 'Matrix'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:S4Vectors':\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'DelayedArray'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n",
      "\n",
      "Loading required package: rhdf5\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# \n",
    "# R Notebook for R Review - BLANK\n",
    "# QBIO Public Data Analysis Student Group\n",
    "# 3/12/2021\n",
    "\n",
    "# Here is a review R Notebook. This is designed to be useful for the small group project. \n",
    "# \n",
    "\n",
    "#if(!requireNamespace(\"BiocManager\", quietly = TRUE)) install.packages(\"BiocManager\")\n",
    "#if(!requireNamespace(\"devtools\")) BiocManager::install(c(\"devtools\"))\n",
    "#if(!requireNamespace(\"robustbase\"))BiocManager::install(c(\"robustbase\"))\n",
    "#library(devtools)\n",
    "#library(robustbase)\n",
    "#devtools::install_github(\"BioinformaticsFMRP/TCGAbiolinks\")\n",
    "if(!requireNamespace(\"SummarizedExperiment\"))BiocManager::install(c(\"SummarizedExperiment\"))\n",
    "if(!requireNamespace(\"HDF5Array\"))BiocManager::install(\"HDF5Array\")\n",
    "library(TCGAbiolinks)\n",
    "library(SummarizedExperiment)\n",
    "library(HDF5Array)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# '''\n",
    "# Use these barcodes if using on your local machine as an example for smaller files downloaded\n",
    "# '''{r}\n",
    "barcodes_rnaseq <- c(\"TCGA-BH-A0DG-01A-21R-A12P-07\",\"TCGA-A2-A0YF-01A-21R-A109-07\",\n",
    "          \"TCGA-AN-A04A-01A-21R-A034-07\",\"TCGA-AR-A1AP-01A-11R-A12P-07\",\n",
    "           \"TCGA-A2-A0T3-01A-21R-A115-07\", \"TCGA-E2-A154-01A-11R-A115-07\" )\n",
    "barcodes_clinic <- c(\"TCGA-BH-A0DG\",\"TCGA-A2-A0YF\",\"TCGA-AN-A04A\",\"TCGA-AR-A1AP\", \"TCGA-A2-A0T3\",\n",
    "                      \"TCGA-E2-A154\", \"TCGA-AO-A12F\", \"TCGA-A2-A0YM\", \"TCGA-BH-A0E0\", \"TCGA-AN-A0FL\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "brca subtype information from:doi.org/10.1016/j.ccell.2018.03.014\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 6 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>patient</th><th scope=col>Tumor.Type</th><th scope=col>Included_in_previous_marker_papers</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>pathologic_stage</th><th scope=col>Tumor_Grade</th><th scope=col>...</th><th scope=col>tobacco_smoking_history</th><th scope=col>CNV Clusters</th><th scope=col>Mutation Clusters</th><th scope=col>DNA.Methylation Clusters</th><th scope=col>mRNA Clusters</th><th scope=col>miRNA Clusters</th><th scope=col>lncRNA Clusters</th><th scope=col>Protein Clusters</th><th scope=col>PARADIGM Clusters</th><th scope=col>Pan-Gyn Clusters</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>...</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>TCGA-3C-AAAU</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-20211</td><td>NA</td><td>4047</td><td>55</td><td>NA       </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C7</td><td>C1</td><td>C1</td><td>C3</td><td>NA</td><td>NA</td><td>C5</td><td>NA</td></tr>\n",
       "\t<tr><td>TCGA-3C-AALI</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-18538</td><td>NA</td><td>4005</td><td>50</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C9</td><td>C2</td><td>C2</td><td>C3</td><td>NA</td><td>C2</td><td>C4</td><td>C4</td></tr>\n",
       "\t<tr><td>TCGA-3C-AALJ</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-22848</td><td>NA</td><td>1474</td><td>62</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C4</td><td>C2</td><td>C2</td><td>C2</td><td>NA</td><td>NA</td><td>C4</td><td>NA</td></tr>\n",
       "\t<tr><td>TCGA-3C-AALK</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-19074</td><td>NA</td><td>1448</td><td>52</td><td>Stage_I  </td><td>NA</td><td>...</td><td>NA</td><td>C1</td><td>C5</td><td>C2</td><td>C2</td><td>C2</td><td>NA</td><td>C2</td><td>C6</td><td>C4</td></tr>\n",
       "\t<tr><td>TCGA-4H-AAAK</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-18371</td><td>NA</td><td>348 </td><td>50</td><td>Stage_III</td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C4</td><td>C1</td><td>C2</td><td>C2</td><td>NA</td><td>C2</td><td>C6</td><td>C1</td></tr>\n",
       "\t<tr><td>TCGA-5L-AAT0</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-15393</td><td>NA</td><td>1477</td><td>42</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C1</td><td>C9</td><td>C1</td><td>C2</td><td>C2</td><td>NA</td><td>NA</td><td>C6</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " patient & Tumor.Type & Included\\_in\\_previous\\_marker\\_papers & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & age\\_at\\_initial\\_pathologic\\_diagnosis & pathologic\\_stage & Tumor\\_Grade & ... & tobacco\\_smoking\\_history & CNV Clusters & Mutation Clusters & DNA.Methylation Clusters & mRNA Clusters & miRNA Clusters & lncRNA Clusters & Protein Clusters & PARADIGM Clusters & Pan-Gyn Clusters\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & ... & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t TCGA-3C-AAAU & BRCA & NO & Alive & -20211 & NA & 4047 & 55 & NA        & NA & ... & NA & C6 & C7 & C1 & C1 & C3 & NA & NA & C5 & NA\\\\\n",
       "\t TCGA-3C-AALI & BRCA & NO & Alive & -18538 & NA & 4005 & 50 & Stage\\_II  & NA & ... & NA & C6 & C9 & C2 & C2 & C3 & NA & C2 & C4 & C4\\\\\n",
       "\t TCGA-3C-AALJ & BRCA & NO & Alive & -22848 & NA & 1474 & 62 & Stage\\_II  & NA & ... & NA & C6 & C4 & C2 & C2 & C2 & NA & NA & C4 & NA\\\\\n",
       "\t TCGA-3C-AALK & BRCA & NO & Alive & -19074 & NA & 1448 & 52 & Stage\\_I   & NA & ... & NA & C1 & C5 & C2 & C2 & C2 & NA & C2 & C6 & C4\\\\\n",
       "\t TCGA-4H-AAAK & BRCA & NO & Alive & -18371 & NA & 348  & 50 & Stage\\_III & NA & ... & NA & C6 & C4 & C1 & C2 & C2 & NA & C2 & C6 & C1\\\\\n",
       "\t TCGA-5L-AAT0 & BRCA & NO & Alive & -15393 & NA & 1477 & 42 & Stage\\_II  & NA & ... & NA & C1 & C9 & C1 & C2 & C2 & NA & NA & C6 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 24\n",
       "\n",
       "| patient &lt;chr&gt; | Tumor.Type &lt;chr&gt; | Included_in_previous_marker_papers &lt;chr&gt; | vital_status &lt;chr&gt; | days_to_birth &lt;chr&gt; | days_to_death &lt;chr&gt; | days_to_last_followup &lt;chr&gt; | age_at_initial_pathologic_diagnosis &lt;dbl&gt; | pathologic_stage &lt;chr&gt; | Tumor_Grade &lt;chr&gt; | ... ... | tobacco_smoking_history &lt;chr&gt; | CNV Clusters &lt;chr&gt; | Mutation Clusters &lt;chr&gt; | DNA.Methylation Clusters &lt;chr&gt; | mRNA Clusters &lt;chr&gt; | miRNA Clusters &lt;chr&gt; | lncRNA Clusters &lt;chr&gt; | Protein Clusters &lt;chr&gt; | PARADIGM Clusters &lt;chr&gt; | Pan-Gyn Clusters &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| TCGA-3C-AAAU | BRCA | NO | Alive | -20211 | NA | 4047 | 55 | NA        | NA | ... | NA | C6 | C7 | C1 | C1 | C3 | NA | NA | C5 | NA |\n",
       "| TCGA-3C-AALI | BRCA | NO | Alive | -18538 | NA | 4005 | 50 | Stage_II  | NA | ... | NA | C6 | C9 | C2 | C2 | C3 | NA | C2 | C4 | C4 |\n",
       "| TCGA-3C-AALJ | BRCA | NO | Alive | -22848 | NA | 1474 | 62 | Stage_II  | NA | ... | NA | C6 | C4 | C2 | C2 | C2 | NA | NA | C4 | NA |\n",
       "| TCGA-3C-AALK | BRCA | NO | Alive | -19074 | NA | 1448 | 52 | Stage_I   | NA | ... | NA | C1 | C5 | C2 | C2 | C2 | NA | C2 | C6 | C4 |\n",
       "| TCGA-4H-AAAK | BRCA | NO | Alive | -18371 | NA | 348  | 50 | Stage_III | NA | ... | NA | C6 | C4 | C1 | C2 | C2 | NA | C2 | C6 | C1 |\n",
       "| TCGA-5L-AAT0 | BRCA | NO | Alive | -15393 | NA | 1477 | 42 | Stage_II  | NA | ... | NA | C1 | C9 | C1 | C2 | C2 | NA | NA | C6 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  patient      Tumor.Type Included_in_previous_marker_papers vital_status\n",
       "1 TCGA-3C-AAAU BRCA       NO                                 Alive       \n",
       "2 TCGA-3C-AALI BRCA       NO                                 Alive       \n",
       "3 TCGA-3C-AALJ BRCA       NO                                 Alive       \n",
       "4 TCGA-3C-AALK BRCA       NO                                 Alive       \n",
       "5 TCGA-4H-AAAK BRCA       NO                                 Alive       \n",
       "6 TCGA-5L-AAT0 BRCA       NO                                 Alive       \n",
       "  days_to_birth days_to_death days_to_last_followup\n",
       "1 -20211        NA            4047                 \n",
       "2 -18538        NA            4005                 \n",
       "3 -22848        NA            1474                 \n",
       "4 -19074        NA            1448                 \n",
       "5 -18371        NA            348                  \n",
       "6 -15393        NA            1477                 \n",
       "  age_at_initial_pathologic_diagnosis pathologic_stage Tumor_Grade ...\n",
       "1 55                                  NA               NA          ...\n",
       "2 50                                  Stage_II         NA          ...\n",
       "3 62                                  Stage_II         NA          ...\n",
       "4 52                                  Stage_I          NA          ...\n",
       "5 50                                  Stage_III        NA          ...\n",
       "6 42                                  Stage_II         NA          ...\n",
       "  tobacco_smoking_history CNV Clusters Mutation Clusters\n",
       "1 NA                      C6           C7               \n",
       "2 NA                      C6           C9               \n",
       "3 NA                      C6           C4               \n",
       "4 NA                      C1           C5               \n",
       "5 NA                      C6           C4               \n",
       "6 NA                      C1           C9               \n",
       "  DNA.Methylation Clusters mRNA Clusters miRNA Clusters lncRNA Clusters\n",
       "1 C1                       C1            C3             NA             \n",
       "2 C2                       C2            C3             NA             \n",
       "3 C2                       C2            C2             NA             \n",
       "4 C2                       C2            C2             NA             \n",
       "5 C1                       C2            C2             NA             \n",
       "6 C1                       C2            C2             NA             \n",
       "  Protein Clusters PARADIGM Clusters Pan-Gyn Clusters\n",
       "1 NA               C5                NA              \n",
       "2 C2               C4                C4              \n",
       "3 NA               C4                NA              \n",
       "4 C2               C6                C4              \n",
       "5 C2               C6                C1              \n",
       "6 NA               C6                NA              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'patient'</li><li>'Tumor.Type'</li><li>'Included_in_previous_marker_papers'</li><li>'vital_status'</li><li>'days_to_birth'</li><li>'days_to_death'</li><li>'days_to_last_followup'</li><li>'age_at_initial_pathologic_diagnosis'</li><li>'pathologic_stage'</li><li>'Tumor_Grade'</li><li>'BRCA_Pathology'</li><li>'BRCA_Subtype_PAM50'</li><li>'MSI_status'</li><li>'HPV_Status'</li><li>'tobacco_smoking_history'</li><li>'CNV Clusters'</li><li>'Mutation Clusters'</li><li>'DNA.Methylation Clusters'</li><li>'mRNA Clusters'</li><li>'miRNA Clusters'</li><li>'lncRNA Clusters'</li><li>'Protein Clusters'</li><li>'PARADIGM Clusters'</li><li>'Pan-Gyn Clusters'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'patient'\n",
       "\\item 'Tumor.Type'\n",
       "\\item 'Included\\_in\\_previous\\_marker\\_papers'\n",
       "\\item 'vital\\_status'\n",
       "\\item 'days\\_to\\_birth'\n",
       "\\item 'days\\_to\\_death'\n",
       "\\item 'days\\_to\\_last\\_followup'\n",
       "\\item 'age\\_at\\_initial\\_pathologic\\_diagnosis'\n",
       "\\item 'pathologic\\_stage'\n",
       "\\item 'Tumor\\_Grade'\n",
       "\\item 'BRCA\\_Pathology'\n",
       "\\item 'BRCA\\_Subtype\\_PAM50'\n",
       "\\item 'MSI\\_status'\n",
       "\\item 'HPV\\_Status'\n",
       "\\item 'tobacco\\_smoking\\_history'\n",
       "\\item 'CNV Clusters'\n",
       "\\item 'Mutation Clusters'\n",
       "\\item 'DNA.Methylation Clusters'\n",
       "\\item 'mRNA Clusters'\n",
       "\\item 'miRNA Clusters'\n",
       "\\item 'lncRNA Clusters'\n",
       "\\item 'Protein Clusters'\n",
       "\\item 'PARADIGM Clusters'\n",
       "\\item 'Pan-Gyn Clusters'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'patient'\n",
       "2. 'Tumor.Type'\n",
       "3. 'Included_in_previous_marker_papers'\n",
       "4. 'vital_status'\n",
       "5. 'days_to_birth'\n",
       "6. 'days_to_death'\n",
       "7. 'days_to_last_followup'\n",
       "8. 'age_at_initial_pathologic_diagnosis'\n",
       "9. 'pathologic_stage'\n",
       "10. 'Tumor_Grade'\n",
       "11. 'BRCA_Pathology'\n",
       "12. 'BRCA_Subtype_PAM50'\n",
       "13. 'MSI_status'\n",
       "14. 'HPV_Status'\n",
       "15. 'tobacco_smoking_history'\n",
       "16. 'CNV Clusters'\n",
       "17. 'Mutation Clusters'\n",
       "18. 'DNA.Methylation Clusters'\n",
       "19. 'mRNA Clusters'\n",
       "20. 'miRNA Clusters'\n",
       "21. 'lncRNA Clusters'\n",
       "22. 'Protein Clusters'\n",
       "23. 'PARADIGM Clusters'\n",
       "24. 'Pan-Gyn Clusters'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"patient\"                             \"Tumor.Type\"                         \n",
       " [3] \"Included_in_previous_marker_papers\"  \"vital_status\"                       \n",
       " [5] \"days_to_birth\"                       \"days_to_death\"                      \n",
       " [7] \"days_to_last_followup\"               \"age_at_initial_pathologic_diagnosis\"\n",
       " [9] \"pathologic_stage\"                    \"Tumor_Grade\"                        \n",
       "[11] \"BRCA_Pathology\"                      \"BRCA_Subtype_PAM50\"                 \n",
       "[13] \"MSI_status\"                          \"HPV_Status\"                         \n",
       "[15] \"tobacco_smoking_history\"             \"CNV Clusters\"                       \n",
       "[17] \"Mutation Clusters\"                   \"DNA.Methylation Clusters\"           \n",
       "[19] \"mRNA Clusters\"                       \"miRNA Clusters\"                     \n",
       "[21] \"lncRNA Clusters\"                     \"Protein Clusters\"                   \n",
       "[23] \"PARADIGM Clusters\"                   \"Pan-Gyn Clusters\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#PART 1: DATAFRAMES\n",
    "\n",
    "# Load in the subtype data\n",
    "# 1. Use head() to look at the subtypes dataframe\n",
    "#   a. What columns are in subtypes? Remember the colnames() function\n",
    "#   b. Which columns do you personally find interesting and why?\n",
    "# {r}\n",
    "subtypes <- TCGAquery_subtype(tumor = \"BRCA\")\n",
    "#### YOUR CODE HERE #####\n",
    "head(subtypes)\n",
    "colnames(subtypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tibble [1,087 x 24] (S3: tbl_df/tbl/data.frame)\n",
      " $ patient                            : chr [1:1087] \"TCGA-3C-AAAU\" \"TCGA-3C-AALI\" \"TCGA-3C-AALJ\" \"TCGA-3C-AALK\" ...\n",
      " $ Tumor.Type                         : chr [1:1087] \"BRCA\" \"BRCA\" \"BRCA\" \"BRCA\" ...\n",
      " $ Included_in_previous_marker_papers : chr [1:1087] \"NO\" \"NO\" \"NO\" \"NO\" ...\n",
      " $ vital_status                       : chr [1:1087] \"Alive\" \"Alive\" \"Alive\" \"Alive\" ...\n",
      " $ days_to_birth                      : chr [1:1087] \"-20211\" \"-18538\" \"-22848\" \"-19074\" ...\n",
      " $ days_to_death                      : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ days_to_last_followup              : chr [1:1087] \"4047\" \"4005\" \"1474\" \"1448\" ...\n",
      " $ age_at_initial_pathologic_diagnosis: num [1:1087] 55 50 62 52 50 42 63 52 70 59 ...\n",
      " $ pathologic_stage                   : chr [1:1087] \"NA\" \"Stage_II\" \"Stage_II\" \"Stage_I\" ...\n",
      " $ Tumor_Grade                        : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ BRCA_Pathology                     : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ BRCA_Subtype_PAM50                 : chr [1:1087] \"LumA\" \"Her2\" \"LumB\" \"LumA\" ...\n",
      " $ MSI_status                         : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ HPV_Status                         : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ tobacco_smoking_history            : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ CNV Clusters                       : chr [1:1087] \"C6\" \"C6\" \"C6\" \"C1\" ...\n",
      " $ Mutation Clusters                  : chr [1:1087] \"C7\" \"C9\" \"C4\" \"C5\" ...\n",
      " $ DNA.Methylation Clusters           : chr [1:1087] \"C1\" \"C2\" \"C2\" \"C2\" ...\n",
      " $ mRNA Clusters                      : chr [1:1087] \"C1\" \"C2\" \"C2\" \"C2\" ...\n",
      " $ miRNA Clusters                     : chr [1:1087] \"C3\" \"C3\" \"C2\" \"C2\" ...\n",
      " $ lncRNA Clusters                    : chr [1:1087] \"NA\" \"NA\" \"NA\" \"NA\" ...\n",
      " $ Protein Clusters                   : chr [1:1087] \"NA\" \"C2\" \"NA\" \"C2\" ...\n",
      " $ PARADIGM Clusters                  : chr [1:1087] \"C5\" \"C4\" \"C4\" \"C6\" ...\n",
      " $ Pan-Gyn Clusters                   : chr [1:1087] \"NA\" \"C4\" \"NA\" \"C4\" ...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1087</li><li>24</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1087\n",
       "\\item 24\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1087\n",
       "2. 24\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1087   24"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# '''\n",
    "# 2. Use str() and dim() to look at subtypes\n",
    "#   a. How is str() different than head?\n",
    "    # Both give column names and types contained in the column. head() does so in a format faithful to the original table\n",
    "    # and provides examples of the first few rows. str() also provides examples, but gives the columns as rows and gives\n",
    "    #dimensions of each column\n",
    "#   b. How many rows and columns are in subtypes?\n",
    "    #1087 rows by 24 columns\n",
    "#   c. How many patients are in subtypes?\n",
    "    #1087 patients\n",
    "# '''{r}\n",
    "#### YOUR CODE HERE #####\n",
    "# '''\n",
    "str(subtypes)\n",
    "dim(subtypes)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 1 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Included_in_previous_marker_papers</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>NO</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 1\n",
       "\\begin{tabular}{l}\n",
       " Included\\_in\\_previous\\_marker\\_papers\\\\\n",
       " <chr>\\\\\n",
       "\\hline\n",
       "\t NO\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 1\n",
       "\n",
       "| Included_in_previous_marker_papers &lt;chr&gt; |\n",
       "|---|\n",
       "| NO |\n",
       "\n"
      ],
      "text/plain": [
       "  Included_in_previous_marker_papers\n",
       "1 NO                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 1 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>patient</th><th scope=col>Tumor.Type</th><th scope=col>Included_in_previous_marker_papers</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>pathologic_stage</th><th scope=col>Tumor_Grade</th><th scope=col>...</th><th scope=col>tobacco_smoking_history</th><th scope=col>CNV Clusters</th><th scope=col>Mutation Clusters</th><th scope=col>DNA.Methylation Clusters</th><th scope=col>mRNA Clusters</th><th scope=col>miRNA Clusters</th><th scope=col>lncRNA Clusters</th><th scope=col>Protein Clusters</th><th scope=col>PARADIGM Clusters</th><th scope=col>Pan-Gyn Clusters</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>...</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>TCGA-5L-AAT1</td><td>BRCA</td><td>NO</td><td>Alive</td><td>-23225</td><td>NA</td><td>1471</td><td>63</td><td>Stage_IV</td><td>NA</td><td>...</td><td>NA</td><td>NA</td><td>C9</td><td>C1</td><td>C2</td><td>NA</td><td>NA</td><td>C2</td><td>NA</td><td>C1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " patient & Tumor.Type & Included\\_in\\_previous\\_marker\\_papers & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & age\\_at\\_initial\\_pathologic\\_diagnosis & pathologic\\_stage & Tumor\\_Grade & ... & tobacco\\_smoking\\_history & CNV Clusters & Mutation Clusters & DNA.Methylation Clusters & mRNA Clusters & miRNA Clusters & lncRNA Clusters & Protein Clusters & PARADIGM Clusters & Pan-Gyn Clusters\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & ... & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t TCGA-5L-AAT1 & BRCA & NO & Alive & -23225 & NA & 1471 & 63 & Stage\\_IV & NA & ... & NA & NA & C9 & C1 & C2 & NA & NA & C2 & NA & C1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 24\n",
       "\n",
       "| patient &lt;chr&gt; | Tumor.Type &lt;chr&gt; | Included_in_previous_marker_papers &lt;chr&gt; | vital_status &lt;chr&gt; | days_to_birth &lt;chr&gt; | days_to_death &lt;chr&gt; | days_to_last_followup &lt;chr&gt; | age_at_initial_pathologic_diagnosis &lt;dbl&gt; | pathologic_stage &lt;chr&gt; | Tumor_Grade &lt;chr&gt; | ... ... | tobacco_smoking_history &lt;chr&gt; | CNV Clusters &lt;chr&gt; | Mutation Clusters &lt;chr&gt; | DNA.Methylation Clusters &lt;chr&gt; | mRNA Clusters &lt;chr&gt; | miRNA Clusters &lt;chr&gt; | lncRNA Clusters &lt;chr&gt; | Protein Clusters &lt;chr&gt; | PARADIGM Clusters &lt;chr&gt; | Pan-Gyn Clusters &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| TCGA-5L-AAT1 | BRCA | NO | Alive | -23225 | NA | 1471 | 63 | Stage_IV | NA | ... | NA | NA | C9 | C1 | C2 | NA | NA | C2 | NA | C1 |\n",
       "\n"
      ],
      "text/plain": [
       "  patient      Tumor.Type Included_in_previous_marker_papers vital_status\n",
       "1 TCGA-5L-AAT1 BRCA       NO                                 Alive       \n",
       "  days_to_birth days_to_death days_to_last_followup\n",
       "1 -23225        NA            1471                 \n",
       "  age_at_initial_pathologic_diagnosis pathologic_stage Tumor_Grade ...\n",
       "1 63                                  Stage_IV         NA          ...\n",
       "  tobacco_smoking_history CNV Clusters Mutation Clusters\n",
       "1 NA                      NA           C9               \n",
       "  DNA.Methylation Clusters mRNA Clusters miRNA Clusters lncRNA Clusters\n",
       "1 C1                       C2            NA             NA             \n",
       "  Protein Clusters PARADIGM Clusters Pan-Gyn Clusters\n",
       "1 C2               NA                C1              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 10 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>patient</th><th scope=col>Tumor.Type</th><th scope=col>Included_in_previous_marker_papers</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>pathologic_stage</th><th scope=col>Tumor_Grade</th><th scope=col>...</th><th scope=col>tobacco_smoking_history</th><th scope=col>CNV Clusters</th><th scope=col>Mutation Clusters</th><th scope=col>DNA.Methylation Clusters</th><th scope=col>mRNA Clusters</th><th scope=col>miRNA Clusters</th><th scope=col>lncRNA Clusters</th><th scope=col>Protein Clusters</th><th scope=col>PARADIGM Clusters</th><th scope=col>Pan-Gyn Clusters</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>...</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>TCGA-3C-AAAU</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-20211</td><td>NA</td><td>4047</td><td>55</td><td>NA       </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C7</td><td>C1</td><td>C1</td><td>C3</td><td>NA</td><td>NA</td><td>C5</td><td>NA</td></tr>\n",
       "\t<tr><td>TCGA-3C-AALI</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-18538</td><td>NA</td><td>4005</td><td>50</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C9</td><td>C2</td><td>C2</td><td>C3</td><td>NA</td><td>C2</td><td>C4</td><td>C4</td></tr>\n",
       "\t<tr><td>TCGA-3C-AALJ</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-22848</td><td>NA</td><td>1474</td><td>62</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C4</td><td>C2</td><td>C2</td><td>C2</td><td>NA</td><td>NA</td><td>C4</td><td>NA</td></tr>\n",
       "\t<tr><td>TCGA-3C-AALK</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-19074</td><td>NA</td><td>1448</td><td>52</td><td>Stage_I  </td><td>NA</td><td>...</td><td>NA</td><td>C1</td><td>C5</td><td>C2</td><td>C2</td><td>C2</td><td>NA</td><td>C2</td><td>C6</td><td>C4</td></tr>\n",
       "\t<tr><td>TCGA-4H-AAAK</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-18371</td><td>NA</td><td>348 </td><td>50</td><td>Stage_III</td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C4</td><td>C1</td><td>C2</td><td>C2</td><td>NA</td><td>C2</td><td>C6</td><td>C1</td></tr>\n",
       "\t<tr><td>TCGA-5L-AAT0</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-15393</td><td>NA</td><td>1477</td><td>42</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C1</td><td>C9</td><td>C1</td><td>C2</td><td>C2</td><td>NA</td><td>NA</td><td>C6</td><td>NA</td></tr>\n",
       "\t<tr><td>TCGA-5L-AAT1</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-23225</td><td>NA</td><td>1471</td><td>63</td><td>Stage_IV </td><td>NA</td><td>...</td><td>NA</td><td>NA</td><td>C9</td><td>C1</td><td>C2</td><td>NA</td><td>NA</td><td>C2</td><td>NA</td><td>C1</td></tr>\n",
       "\t<tr><td>TCGA-5T-A9QA</td><td>BRCA</td><td>NO </td><td>Alive</td><td>-19031</td><td>NA</td><td>303 </td><td>52</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C6</td><td>C6</td><td>C2</td><td>C2</td><td>C3</td><td>NA</td><td>C2</td><td>C8</td><td>C5</td></tr>\n",
       "\t<tr><td>TCGA-A1-A0SB</td><td>BRCA</td><td>YES</td><td>Alive</td><td>-25833</td><td>NA</td><td>259 </td><td>70</td><td>Stage_I  </td><td>NA</td><td>...</td><td>NA</td><td>C2</td><td>C4</td><td>C3</td><td>C4</td><td>C4</td><td>NA</td><td>NA</td><td>C6</td><td>NA</td></tr>\n",
       "\t<tr><td>TCGA-A1-A0SD</td><td>BRCA</td><td>YES</td><td>Alive</td><td>-21793</td><td>NA</td><td>437 </td><td>59</td><td>Stage_II </td><td>NA</td><td>...</td><td>NA</td><td>C4</td><td>C4</td><td>C1</td><td>C1</td><td>C3</td><td>NA</td><td>NA</td><td>C6</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 10 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " patient & Tumor.Type & Included\\_in\\_previous\\_marker\\_papers & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & age\\_at\\_initial\\_pathologic\\_diagnosis & pathologic\\_stage & Tumor\\_Grade & ... & tobacco\\_smoking\\_history & CNV Clusters & Mutation Clusters & DNA.Methylation Clusters & mRNA Clusters & miRNA Clusters & lncRNA Clusters & Protein Clusters & PARADIGM Clusters & Pan-Gyn Clusters\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & ... & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t TCGA-3C-AAAU & BRCA & NO  & Alive & -20211 & NA & 4047 & 55 & NA        & NA & ... & NA & C6 & C7 & C1 & C1 & C3 & NA & NA & C5 & NA\\\\\n",
       "\t TCGA-3C-AALI & BRCA & NO  & Alive & -18538 & NA & 4005 & 50 & Stage\\_II  & NA & ... & NA & C6 & C9 & C2 & C2 & C3 & NA & C2 & C4 & C4\\\\\n",
       "\t TCGA-3C-AALJ & BRCA & NO  & Alive & -22848 & NA & 1474 & 62 & Stage\\_II  & NA & ... & NA & C6 & C4 & C2 & C2 & C2 & NA & NA & C4 & NA\\\\\n",
       "\t TCGA-3C-AALK & BRCA & NO  & Alive & -19074 & NA & 1448 & 52 & Stage\\_I   & NA & ... & NA & C1 & C5 & C2 & C2 & C2 & NA & C2 & C6 & C4\\\\\n",
       "\t TCGA-4H-AAAK & BRCA & NO  & Alive & -18371 & NA & 348  & 50 & Stage\\_III & NA & ... & NA & C6 & C4 & C1 & C2 & C2 & NA & C2 & C6 & C1\\\\\n",
       "\t TCGA-5L-AAT0 & BRCA & NO  & Alive & -15393 & NA & 1477 & 42 & Stage\\_II  & NA & ... & NA & C1 & C9 & C1 & C2 & C2 & NA & NA & C6 & NA\\\\\n",
       "\t TCGA-5L-AAT1 & BRCA & NO  & Alive & -23225 & NA & 1471 & 63 & Stage\\_IV  & NA & ... & NA & NA & C9 & C1 & C2 & NA & NA & C2 & NA & C1\\\\\n",
       "\t TCGA-5T-A9QA & BRCA & NO  & Alive & -19031 & NA & 303  & 52 & Stage\\_II  & NA & ... & NA & C6 & C6 & C2 & C2 & C3 & NA & C2 & C8 & C5\\\\\n",
       "\t TCGA-A1-A0SB & BRCA & YES & Alive & -25833 & NA & 259  & 70 & Stage\\_I   & NA & ... & NA & C2 & C4 & C3 & C4 & C4 & NA & NA & C6 & NA\\\\\n",
       "\t TCGA-A1-A0SD & BRCA & YES & Alive & -21793 & NA & 437  & 59 & Stage\\_II  & NA & ... & NA & C4 & C4 & C1 & C1 & C3 & NA & NA & C6 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 10 × 24\n",
       "\n",
       "| patient &lt;chr&gt; | Tumor.Type &lt;chr&gt; | Included_in_previous_marker_papers &lt;chr&gt; | vital_status &lt;chr&gt; | days_to_birth &lt;chr&gt; | days_to_death &lt;chr&gt; | days_to_last_followup &lt;chr&gt; | age_at_initial_pathologic_diagnosis &lt;dbl&gt; | pathologic_stage &lt;chr&gt; | Tumor_Grade &lt;chr&gt; | ... ... | tobacco_smoking_history &lt;chr&gt; | CNV Clusters &lt;chr&gt; | Mutation Clusters &lt;chr&gt; | DNA.Methylation Clusters &lt;chr&gt; | mRNA Clusters &lt;chr&gt; | miRNA Clusters &lt;chr&gt; | lncRNA Clusters &lt;chr&gt; | Protein Clusters &lt;chr&gt; | PARADIGM Clusters &lt;chr&gt; | Pan-Gyn Clusters &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| TCGA-3C-AAAU | BRCA | NO  | Alive | -20211 | NA | 4047 | 55 | NA        | NA | ... | NA | C6 | C7 | C1 | C1 | C3 | NA | NA | C5 | NA |\n",
       "| TCGA-3C-AALI | BRCA | NO  | Alive | -18538 | NA | 4005 | 50 | Stage_II  | NA | ... | NA | C6 | C9 | C2 | C2 | C3 | NA | C2 | C4 | C4 |\n",
       "| TCGA-3C-AALJ | BRCA | NO  | Alive | -22848 | NA | 1474 | 62 | Stage_II  | NA | ... | NA | C6 | C4 | C2 | C2 | C2 | NA | NA | C4 | NA |\n",
       "| TCGA-3C-AALK | BRCA | NO  | Alive | -19074 | NA | 1448 | 52 | Stage_I   | NA | ... | NA | C1 | C5 | C2 | C2 | C2 | NA | C2 | C6 | C4 |\n",
       "| TCGA-4H-AAAK | BRCA | NO  | Alive | -18371 | NA | 348  | 50 | Stage_III | NA | ... | NA | C6 | C4 | C1 | C2 | C2 | NA | C2 | C6 | C1 |\n",
       "| TCGA-5L-AAT0 | BRCA | NO  | Alive | -15393 | NA | 1477 | 42 | Stage_II  | NA | ... | NA | C1 | C9 | C1 | C2 | C2 | NA | NA | C6 | NA |\n",
       "| TCGA-5L-AAT1 | BRCA | NO  | Alive | -23225 | NA | 1471 | 63 | Stage_IV  | NA | ... | NA | NA | C9 | C1 | C2 | NA | NA | C2 | NA | C1 |\n",
       "| TCGA-5T-A9QA | BRCA | NO  | Alive | -19031 | NA | 303  | 52 | Stage_II  | NA | ... | NA | C6 | C6 | C2 | C2 | C3 | NA | C2 | C8 | C5 |\n",
       "| TCGA-A1-A0SB | BRCA | YES | Alive | -25833 | NA | 259  | 70 | Stage_I   | NA | ... | NA | C2 | C4 | C3 | C4 | C4 | NA | NA | C6 | NA |\n",
       "| TCGA-A1-A0SD | BRCA | YES | Alive | -21793 | NA | 437  | 59 | Stage_II  | NA | ... | NA | C4 | C4 | C1 | C1 | C3 | NA | NA | C6 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "   patient      Tumor.Type Included_in_previous_marker_papers vital_status\n",
       "1  TCGA-3C-AAAU BRCA       NO                                 Alive       \n",
       "2  TCGA-3C-AALI BRCA       NO                                 Alive       \n",
       "3  TCGA-3C-AALJ BRCA       NO                                 Alive       \n",
       "4  TCGA-3C-AALK BRCA       NO                                 Alive       \n",
       "5  TCGA-4H-AAAK BRCA       NO                                 Alive       \n",
       "6  TCGA-5L-AAT0 BRCA       NO                                 Alive       \n",
       "7  TCGA-5L-AAT1 BRCA       NO                                 Alive       \n",
       "8  TCGA-5T-A9QA BRCA       NO                                 Alive       \n",
       "9  TCGA-A1-A0SB BRCA       YES                                Alive       \n",
       "10 TCGA-A1-A0SD BRCA       YES                                Alive       \n",
       "   days_to_birth days_to_death days_to_last_followup\n",
       "1  -20211        NA            4047                 \n",
       "2  -18538        NA            4005                 \n",
       "3  -22848        NA            1474                 \n",
       "4  -19074        NA            1448                 \n",
       "5  -18371        NA            348                  \n",
       "6  -15393        NA            1477                 \n",
       "7  -23225        NA            1471                 \n",
       "8  -19031        NA            303                  \n",
       "9  -25833        NA            259                  \n",
       "10 -21793        NA            437                  \n",
       "   age_at_initial_pathologic_diagnosis pathologic_stage Tumor_Grade ...\n",
       "1  55                                  NA               NA          ...\n",
       "2  50                                  Stage_II         NA          ...\n",
       "3  62                                  Stage_II         NA          ...\n",
       "4  52                                  Stage_I          NA          ...\n",
       "5  50                                  Stage_III        NA          ...\n",
       "6  42                                  Stage_II         NA          ...\n",
       "7  63                                  Stage_IV         NA          ...\n",
       "8  52                                  Stage_II         NA          ...\n",
       "9  70                                  Stage_I          NA          ...\n",
       "10 59                                  Stage_II         NA          ...\n",
       "   tobacco_smoking_history CNV Clusters Mutation Clusters\n",
       "1  NA                      C6           C7               \n",
       "2  NA                      C6           C9               \n",
       "3  NA                      C6           C4               \n",
       "4  NA                      C1           C5               \n",
       "5  NA                      C6           C4               \n",
       "6  NA                      C1           C9               \n",
       "7  NA                      NA           C9               \n",
       "8  NA                      C6           C6               \n",
       "9  NA                      C2           C4               \n",
       "10 NA                      C4           C4               \n",
       "   DNA.Methylation Clusters mRNA Clusters miRNA Clusters lncRNA Clusters\n",
       "1  C1                       C1            C3             NA             \n",
       "2  C2                       C2            C3             NA             \n",
       "3  C2                       C2            C2             NA             \n",
       "4  C2                       C2            C2             NA             \n",
       "5  C1                       C2            C2             NA             \n",
       "6  C1                       C2            C2             NA             \n",
       "7  C1                       C2            NA             NA             \n",
       "8  C2                       C2            C3             NA             \n",
       "9  C3                       C4            C4             NA             \n",
       "10 C1                       C1            C3             NA             \n",
       "   Protein Clusters PARADIGM Clusters Pan-Gyn Clusters\n",
       "1  NA               C5                NA              \n",
       "2  C2               C4                C4              \n",
       "3  NA               C4                NA              \n",
       "4  C2               C6                C4              \n",
       "5  C2               C6                C1              \n",
       "6  NA               C6                NA              \n",
       "7  C2               NA                C1              \n",
       "8  C2               C8                C5              \n",
       "9  NA               C6                NA              \n",
       "10 NA               C6                NA              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# 3. Standard Indexing\n",
    "#   a. Use the format, dataframe[row, col] to find the element at row 4 and col 3.\n",
    "#   b. Removing the row or col in [row,col] allows you to access an entire row or column. Access all columns of the 7th row. \n",
    "#   c. 1:10 is a special R way to create a list of from 1 to 10. Type 1:10 in the console if this is unclear. \n",
    "#      Access all columns of the first 10 rows. Note this is the first 10 patients.\n",
    "#   d. Is the format subtypes[row,col] useful for this dataframe? Why or why not?\n",
    "#   e. Don't forget about the $. Create a variable called BRCA_PAM50 and assign the BRCA_Subtype_PAM50 column to it. \n",
    "# '''{r}\n",
    "#### YOUR CODE HERE #####\n",
    "# '''\n",
    "#a\n",
    "subtypes[4, 3]\n",
    "\n",
    "#b\n",
    "subtypes[7,]\n",
    "\n",
    "#c\n",
    "subtypes[1:10,]\n",
    "\n",
    "#d\n",
    "#Yes, provides slices of information\n",
    "\n",
    "#e\n",
    "BRCA_PAM50 <- subtypes$BRCA_Subtype_PAM50\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "76"
      ],
      "text/latex": [
       "76"
      ],
      "text/markdown": [
       "76"
      ],
      "text/plain": [
       "[1] 76"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "989"
      ],
      "text/latex": [
       "989"
      ],
      "text/markdown": [
       "989"
      ],
      "text/plain": [
       "[1] 989"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in cbind(parts$left, ellip_h, parts$right, deparse.level = 0L):\n",
      "\"number of rows of result is not a multiple of vector length (arg 2)\"\n",
      "Warning message in cbind(parts$left, ellip_h, parts$right, deparse.level = 0L):\n",
      "\"number of rows of result is not a multiple of vector length (arg 2)\"\n",
      "Warning message in cbind(parts$left, ellip_h, parts$right, deparse.level = 0L):\n",
      "\"number of rows of result is not a multiple of vector length (arg 2)\"\n",
      "Warning message in cbind(parts$left, ellip_h, parts$right, deparse.level = 0L):\n",
      "\"number of rows of result is not a multiple of vector length (arg 2)\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 0 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>patient</th><th scope=col>Tumor.Type</th><th scope=col>Included_in_previous_marker_papers</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>pathologic_stage</th><th scope=col>Tumor_Grade</th><th scope=col>...</th><th scope=col>tobacco_smoking_history</th><th scope=col>CNV Clusters</th><th scope=col>Mutation Clusters</th><th scope=col>DNA.Methylation Clusters</th><th scope=col>mRNA Clusters</th><th scope=col>miRNA Clusters</th><th scope=col>lncRNA Clusters</th><th scope=col>Protein Clusters</th><th scope=col>PARADIGM Clusters</th><th scope=col>Pan-Gyn Clusters</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>...</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 0 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " patient & Tumor.Type & Included\\_in\\_previous\\_marker\\_papers & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & age\\_at\\_initial\\_pathologic\\_diagnosis & pathologic\\_stage & Tumor\\_Grade & ... & tobacco\\_smoking\\_history & CNV Clusters & Mutation Clusters & DNA.Methylation Clusters & mRNA Clusters & miRNA Clusters & lncRNA Clusters & Protein Clusters & PARADIGM Clusters & Pan-Gyn Clusters\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & ... & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 0 × 24\n",
       "\n",
       "| patient &lt;chr&gt; | Tumor.Type &lt;chr&gt; | Included_in_previous_marker_papers &lt;chr&gt; | vital_status &lt;chr&gt; | days_to_birth &lt;chr&gt; | days_to_death &lt;chr&gt; | days_to_last_followup &lt;chr&gt; | age_at_initial_pathologic_diagnosis &lt;dbl&gt; | pathologic_stage &lt;chr&gt; | Tumor_Grade &lt;chr&gt; | ... ... | tobacco_smoking_history &lt;chr&gt; | CNV Clusters &lt;chr&gt; | Mutation Clusters &lt;chr&gt; | DNA.Methylation Clusters &lt;chr&gt; | mRNA Clusters &lt;chr&gt; | miRNA Clusters &lt;chr&gt; | lncRNA Clusters &lt;chr&gt; | Protein Clusters &lt;chr&gt; | PARADIGM Clusters &lt;chr&gt; | Pan-Gyn Clusters &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     patient Tumor.Type Included_in_previous_marker_papers vital_status\n",
       "     days_to_birth days_to_death days_to_last_followup\n",
       "     age_at_initial_pathologic_diagnosis pathologic_stage Tumor_Grade ...\n",
       "     tobacco_smoking_history CNV Clusters Mutation Clusters\n",
       "     DNA.Methylation Clusters mRNA Clusters miRNA Clusters lncRNA Clusters\n",
       "     Protein Clusters PARADIGM Clusters Pan-Gyn Clusters"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "# 4. Boolean Indexing\n",
    "# Boolean Indexing is a unique and helpful way to index into a dataframe.\n",
    "# It utilizes a \"boolean mask\". A boolean is a TRUE or FALSE element. A boolean mask is a list of TRUE's and FALSE's.\n",
    "# Follow the following steps to find patients less than 40yrs old AND are in the LumA subtype\n",
    "\n",
    "#   1. Describe the contents of luma_mask. Note that \"==\" means \"equal to\"\n",
    "#   2. Write an equivalent line of code that creates the variable luma_mask_check that is equivalent to luma_mask,\n",
    "#      but uses the BRCA_PAM50 variable above.\n",
    "#   3. Computers read TRUE as \"1\" and FALSE as \"0\". A cool trick is to sum the elements of the mask to see how many TRUE elements exist.\n",
    "#      Using sum() and as.numeric(luma_mask), how many patients are Luminal A subtype?\n",
    "#   4. Use information from ^^ check that luma_mask and luma_mask_check have the same number of TRUE elements.\n",
    "#   5. Create a boolean mask called young_mask that is TRUE when the patient is less than (<) 40 yrs old. How many patients are under 40?\n",
    "#      How many patients are over 40 in subtype?\n",
    "#   6. Create a boolean mask called young_and_luma_mask that is TRUE when the patient in LumA and young. Use the \"&\" symbol.\n",
    "#   7. You can use the boolean mask to index into the subtypes dataframe. Remember part 3. Standard Indexing. \n",
    "#      Create a new dataframe called young_and_luma_dataframe and assign it to all columns of the patients that are young and luminal a\n",
    "#      using your young_and_luma_mask. Hint: This is the same format as the 3c using the 1:10. \n",
    "#   8. What if we just want which patients are young and luminal a? You can also index into a specific column. \n",
    "#      Using $ and subtypes[young_and_luma_mask], create a variable young_and_luma_patients with the list of young and luminal a patients.\n",
    "# '''{r}\n",
    "#### YOUR CODE HERE #####\n",
    "# '''\n",
    "#\n",
    "#1\n",
    "luma_mask <- subtypes$BRCA_Subtype_PAM50 == \"LumA\"\n",
    "\n",
    "#2\n",
    "luma_mask_check <- BRCA_PAM50 == \"LumA\"\n",
    "\n",
    "#3\n",
    "#sum(luma_mask)\n",
    "#as.numeric(luma_mask)\n",
    "#sum(as.numeric(luma_mask))\n",
    "\n",
    "#4\n",
    "#sum(luma_mask_check)\n",
    "#as.numeric(luma_mask_check)\n",
    "#sum(as.numeric(luma_mask_check))\n",
    "\n",
    "#5\n",
    "young_mask <- subtypes$age_at_initial_pathologic_diagnosis < 40\n",
    "sum(as.numeric(young_mask))\n",
    "old_mask <- subtypes$age_at_initial_pathologic_diagnosis > 40\n",
    "sum(as.numeric(old_mask))\n",
    "\n",
    "#6\n",
    "#some trouble here\n",
    "young_and_luma_mask <- (young_mask & old_mask)\n",
    "sum(young_and_luma_mask)\n",
    "#7\n",
    "young_and_luma_dataframe <- subtypes[young_and_luma_mask, ]\n",
    "head(young_and_luma_dataframe)\n",
    "#8\n",
    "young_and_luma_patients <- subtypes[young_and_luma_mask, ]$patient\n",
    "\n",
    "length(young_and_luma_patients)\n",
    "# 4.5. Use masks to summarize data.\n",
    "#   a. What percentage of the young (<40) patients in the subtypes data frame are Luminal A?\n",
    "#   b. What percentage of the old (>=60) patients in the subtypes data frame are Luminal A?\n",
    "# '''{r}\n",
    "#### YOUR CODE HERE #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# '''\n",
    "# PART 2: RNASeq data\n",
    "\n",
    "# Download and/or access data\n",
    "# '''{r}\n",
    "query <- GDCquery(project = \"TCGA-BRCA\",\n",
    "                   data.category = \"Transcriptome Profiling\",\n",
    "                   data.type = \"Gene Expression Quantification\",\n",
    "                   workflow.type = \"HTSeq - Counts\",\n",
    "                   barcode = barcodes_rnaseq)\n",
    "GDCdownload(query) #only need this line of code ONCE to download the data\n",
    "sum_exp <- GDCprepare(query)\n",
    "# '''\n",
    "\n",
    "# 1. SummarizedExperiment \n",
    "#    a. What does sum_exp include?\n",
    "#    b. What does colData(sum_exp) include?\n",
    "#    c. What does rowData(sum_exp) include?\n",
    "#    d. What does assays(sum_exp) include?\n",
    "#    e. Describe assays(sum_exp)$\"HTSeq - Counts\"\n",
    "# '''{r}\n",
    "#### YOUR CODE HERE #####\n",
    "# '''\n",
    "\n",
    "# 2. ifelse\n",
    "# Remember the ifelse structure:\n",
    "# ifelse(condition, action when condition true, action when condition false)\n",
    "#    a.Create a new column in colData(sum_exp) called age_category with the following age categories:\n",
    "#      < 40 is \"young\", 40-59 is \"mid\", 60+ is \"old\".\n",
    "#      Use colData(sum_exp)\n",
    "# '''r}\n",
    "#### YOUR CODE HERE #####\n",
    "# '''\n",
    "\n",
    "# 3. Applying boolean masks to HTSeq\n",
    "#    a. Create variable young_htseq_mask using colData(sum_exp)$age_category\n",
    "#    b. Create variable htseq_data that consists of: assays(sum_exp)$\"HTSeq - Counts\"\n",
    "#    c. Create variable young_htseq_data that consists of all genes (rows) of only the young patients (cols). Run str(young_htseq_data).\n",
    "#    d. Use dim(young_htseq_data)[2] == sum(as.numeric()) to check that your code is correct.\n",
    "#    e. Why does (d) verify if your code is correct?\n",
    "# '''{r}\n",
    "#### YOUR CODE HERE #####\n",
    "# '''\n",
    "\n",
    "# 4. Note about barcodes\n",
    "# Notice that in colnames(colData(sum_exp)) there are two columns that discuss barcodes: \n",
    "#                                                     colData(sum_exp)$barcode and colData(sum_exp)$bcr_patient_barcode\n",
    "# Notice the barcode is longer with extra numbers and letters. This is because it refers the the tumor SAMPLE.\n",
    "# The bcr_patient_barcode refers to the PATIENT. You can imagine that one PATIENT can have multiple tumor SAMPLES.\n",
    "# The bcr_patient_barcode is the same in the clinical data, and the subtypes dataframe above. \n",
    "# Alternatively, you can use the colData(sum_exp)$paper_BRCA_Subtype_PAM50 column in sum_exp\n",
    "# '''{r}\n",
    "head( colData(sum_exp)$barcode )\n",
    "head( colData(sum_exp)$bcr_patient_barcode )\n",
    "head( colData(sum_exp)$paper_BRCA_Subtype_PAM50 )\n",
    "# '''\n",
    "\n",
    "# 5. Note about saving the SummarizedExperiment\n",
    "# As you perform your analysis, you can reload in your TCGA HTSeq data through the query function as we have been doing. \n",
    "# Alternatively, you can do the query once, and save the SummarizedExperiment as an HDF5Array\n",
    "#    1. Use getwd() to print out your current PATH. This is where you are currently located in your system. \n",
    "#    2. Whenever you are saving or loading in files, you can use setwd() to specify the PATH you wish to reference. \n",
    "#    3. Use saveHDF5SummarizedExperiment() to save your SummarizedExperiment in desired directory (dir). Remember, this is RELATIVE to your current location or PATH that you just specified in setwd(), or looked at in getwd(). \n",
    "   \n",
    "# Source: https://bioconductor.org/packages/release/bioc/manuals/HDF5Array/man/HDF5Array.pdf\n",
    "\n",
    "# '''{r}\n",
    "getwd()\n",
    "setwd(\"C:/Users/Kate/qbio_data_analysis/\")\n",
    "#the dir \n",
    "saveHDF5SummarizedExperiment(sum_exp, dir=\"htseq_h5_sumexp\", prefix=\"\", replace=FALSE,chunkdim=NULL, level=NULL, as.sparse=NA,verbose=NA)\n",
    "#To load in your hdf5 summarizedexperiment\n",
    "sum_exp_loaded <- loadHDF5SummarizedExperiment(dir=\"htseq_h5_sumexp\", prefix=\"\")\n",
    "str(sum_exp_loaded)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
